analytical_engine/core/java/graphx/local_vertex_map.h

```diff
@@ -0,0 +1,237 @@
+    vertex_map->meta_.SetNBytes(nbytes);
+
+    VINEYARD_CHECK_OK(
+        client.CreateMetaData(vertex_map->meta_, vertex_map->id_));
+    // mark the builder as sealed
+    this->set_sealed(true);
+
+    return std::static_pointer_cast<vineyard::Object>(vertex_map);
+  }
+
+ private:
+  vineyard::Client& client_;
+  vineyard_oid_array_t inner_lid2Oid_, outer_lid2Oid_;
+  vineyard::NumericArray<int32_t> pid_array_;
+};
+
+template <typename OID_T, typename VID_T>
+class BasicLocalVertexMapBuilder : public LocalVertexMapBuilder<OID_T, VID_T> {
+  using oid_t = OID_T;
+  using vid_t = VID_T;
+  using oid_array_t = typename vineyard::ConvertToArrowType<oid_t>::ArrayType;
+  using oid_array_builder_t =
+      typename vineyard::ConvertToArrowType<oid_t>::BuilderType;
+  using pid_array_builder_t =
+      typename vineyard::ConvertToArrowType<int32_t>::BuilderType;
+
+ public:
+  BasicLocalVertexMapBuilder(vineyard::Client& client,
+                             oid_array_builder_t& inner_oids_builder,
+                             oid_array_builder_t& outer_oids_builder,
+                             pid_array_builder_t& pid_array_builder)
+      : LocalVertexMapBuilder<oid_t, vid_t>(client) {
+    CHECK(inner_oids_builder.Finish(&inner_oids_).ok());
+    CHECK(outer_oids_builder.Finish(&outer_oids_).ok());
+    CHECK(pid_array_builder.Finish(&pid_array_).ok());
+  }
+
+  vineyard::Status Build(vineyard::Client& client) override {
+#if defined(WITH_PROFILING)
+    auto start_ts = grape::GetCurrentTime();
+#endif
+
+    typename vineyard::InternalType<oid_t>::vineyard_builder_type
+        inner_array_builder(client, inner_oids_);
+    this->SetInnerOidArray(
+        *std::dynamic_pointer_cast<vineyard::NumericArray<oid_t>>(
+            inner_array_builder.Seal(client)));
+    typename vineyard::InternalType<oid_t>::vineyard_builder_type
+        outer_array_builder(client, outer_oids_);
+    this->SetOuterOidArray(
+        *std::dynamic_pointer_cast<vineyard::NumericArray<oid_t>>(
+            outer_array_builder.Seal(client)));
+
+    typename vineyard::InternalType<int32_t>::vineyard_builder_type
+        pid_array_builder(client, pid_array_);
+    this->SetPidArray(
+        *std::dynamic_pointer_cast<vineyard::NumericArray<int32_t>>(
+            pid_array_builder.Seal(client)));
+    LOG(INFO) << "Finish setting inner and outer oids";
+
+#if defined(WITH_PROFILING)
+    auto finish_seal_ts = grape::GetCurrentTime();
+    LOG(INFO) << "Seal hashmaps uses " << (finish_seal_ts - start_ts)
+              << " seconds";
+#endif
+    return vineyard::Status::OK();
+  }
+  std::shared_ptr<LocalVertexMap<oid_t, vid_t>> MySeal(
+      vineyard::Client& client) {
+    return std::dynamic_pointer_cast<LocalVertexMap<oid_t, vid_t>>(
+        this->Seal(client));
+  }
+
+ private:
+  std::shared_ptr<oid_array_t> inner_oids_, outer_oids_;
+  std::shared_ptr<arrow::Int32Array> pid_array_;
+};
+
+}  // namespace gs
```

Comment:
It seems that the local vertex map has some level of overlap with @acezen's previous work. Could you folks have some collaboration on that?